### ISSUE TITLE
ORM second level cache and version (locking)

### ISSUE THREAD

**Issue by wtorsi** [doctrine/doctrine2#7067]
Hi, 
I have found strange behaviour when using Second Level Cache and Version.
When new version is set second level cache does not include it into update.

After some investigation:
The problem occurs here `DefaultEntityHydrator::buildCacheEntry`, UnitOfWork does not include version field in actualData.

Any ideas how to fix it?

**Comment by Ocramius** [comment]
This is way too little information to work on anything. Please try re-creating a failure with an isolated test. See examples at https://github.com/doctrine/doctrine2/tree/master/tests/Doctrine/Tests/ORM/Functional/Ticket

**Comment by wtorsi** [comment]
I have never done it before. 
Also, i'm using Symfony Bridge, I have never configured Doctrine from the box :) 
Ok, i'll try. 

Here is the fast fix:


```
// Doctrine\ORM\Cache\DefaultEntityHydrator::buildCacheEntry

$data = $this->uow->getOriginalEntityData($entity);
$data = array_merge($data, $metadata->getIdentifierValues($entity)); // why update has no identifier values ?

//here it is
if ($metadata->isVersioned) {
    $data[$metadata->versionField] = $metadata->getFieldValue($entity, $metadata->versionField);
}

```

This can fix the problem.

**Comment by wtorsi** [comment]
@Ocramius hi again, I need some help.
How I can push a commit with test to repo? It's my first time, sorry.

**Comment by Ocramius** [comment]
@wtorsi you first need to fork this repository (top right on this page), then follow https://yangsu.github.io/pull-request-tutorial/ 👍

**Comment by wtorsi** [comment]
Thank's

**Comment by wtorsi** [comment]
Got it, here is the test.
Here is the test and fix for doctrine 3

**Comment by lcobucci** [comment]
Fixed by #7069
